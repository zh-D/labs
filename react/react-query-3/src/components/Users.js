import { useQuery } from 'react-query'

const Users = () => {
    const { isLoading, isFetching, error, data, status } = useQuery('Users', () =>
        fetch('https://jsonplaceholder.typicode.com/users').then(res =>
            res.json()
        )
    )

    console.log(isFetching);
    console.log(isLoading);
    console.log(error);
    console.log(data);

    if (isLoading) {
        return <p>Loading...</p>
    }

    return (
        <div>
            <h1>Welcome to React Query 3</h1>
            {isFetching && "Background Updating"}
            {/* {isLoading && "Loading..."} */}
            {error && error.message}
            {data && data.map(user => <h1 key={user.id}>{user.name}</h1>)}
        </div>
    )
}

export default Users