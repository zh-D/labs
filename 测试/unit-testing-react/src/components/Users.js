import React from 'react'

const Users = ({ user }) => {

    const legal = user.age >= 18 ? <h1 style={{ color: "green" }}>{user.name}</h1> : <h1 style={{ color: "red" }}>{user.name}</h1>

    return (
        <div data-testid={`user-${user.id}`}>
            {legal}
        </div>
    )
}

export default Users
