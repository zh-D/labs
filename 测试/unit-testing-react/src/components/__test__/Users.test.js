import { render, screen, cleanup } from '@testing-library/react'
import Users from '../Users'
import renderer from "react-test-renderer"

// test("should render Users component", () => {
//     render(<Users />) 
//     const userElement = screen.getByTestId("user-1")
//     expect(userElement).toBeInTheDocument()
//     expect(userElement).toHaveTextContent("Hi")
// })

afterEach(() => {
    cleanup()
})

test("Should render illegal user", () => {
    const user = { id: 1, name: "John", age: 15 }

    render(<Users user={user} />)

    const userElement = screen.getByTestId("user-1")
    expect(userElement).toBeInTheDocument()
    expect(userElement).toHaveTextContent("John")

    expect(userElement).toContainHTML("red")
})

test("matches snapshots", () => {
    const user = { id: 1, name: "John", age: 15 }
    const tree = renderer.create(<Users user={user} />).toJSON()
    expect(tree).toMatchSnapshot()
})