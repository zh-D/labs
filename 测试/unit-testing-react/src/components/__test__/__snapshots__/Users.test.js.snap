// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  data-testid="user-1"
>
  <h1
    style={
      Object {
        "color": "red",
      }
    }
  >
    John
  </h1>
</div>
`;
